contrib/platform/src/com/sun/jna/platform/mac/IOKit.java

```diff
@@ -0,0 +1,807 @@
+         *         released by the caller when it is finished.
+         */
+        public IORegistryEntry next() {
+            return INSTANCE.IOIteratorNext(this);
+        }
+    }
+
+    /**
+     * The base class for all objects in the registry.
+     */
+    class IORegistryEntry extends IOObject {
+        public IORegistryEntry() {
+            super();
+        }
+
+        public IORegistryEntry(Pointer p) {
+            super(p);
+        }
+
+        /**
+         * Convenience method for {@link #IORegistryEntryGetRegistryEntryID} to return
+         * an ID for this registry entry that is global to all tasks.
+         *
+         * @param id
+         *            The resulting ID.
+         * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+         */
+        public int getRegistryEntryID(LongByReference id) {
+            return INSTANCE.IORegistryEntryGetRegistryEntryID(this, id);
+        }
+
+        /**
+         * Convenience method for {@link #IORegistryEntryGetName} to return a name
+         * assigned to this registry entry.
+         *
+         * @param name
+         *            The caller's buffer to receive the name. This must be a 128-byte
+         *            buffer.
+         * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+         */
+        public int getName(Pointer name) {
+            return INSTANCE.IORegistryEntryGetName(this, name);
+        }
+
+        /**
+         * Convenience method for {@link #IORegistryEntryGetChildIterator} to return an
+         * iterator over this registry entry’s child entries in a plane.
+         *
+         * @param plane
+         *            The name of an existing registry plane. Plane names are defined in
+         *            {@code IOKitKeys.h}, for example, {@code kIOServicePlane}.
+         * @param iter
+         *            The created iterator over the children of the entry, on success.
+         *            The iterator must be released when the iteration is finished.
+         * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+         */
+        public int getChildIterator(String plane, PointerByReference iter) {
+            return INSTANCE.IORegistryEntryGetChildIterator(this, plane, iter);
+        }
+
+        /**
+         * Convenience method for {@link #IORegistryEntryGetChildEntry} to return the
+         * first child of this registry entry in a plane.
+         *
+         * @param plane
+         *            The name of an existing registry plane.
+         * @param child
+         *            The first child of the registry entry, on success. The child must
+         *            be released by the caller.
+         * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+         */
+        public int getChildEntry(String plane, PointerByReference child) {
+            return INSTANCE.IORegistryEntryGetChildEntry(this, plane, child);
```

Comment:
Same comment as in `getRegistryEntryID`. In this case an `IORegistryEntry` could be returned directly.

Reply:
Actually, this may fail commonly if there are no children.  Might have to check for that (expected) return and return null instead.